react_app/src/pages/MovieListPage/MovieListPage.jsx

```diff
@@ -2,63 +2,69 @@ import { useEffect, useState } from "react";
 
 import { FILMS_DATA, genres } from "../../data/data";
 import GenreSelect from "../../components/GenreSelect/GenreSelect";
-import SortControl, {
-  TYPE_SORT_NAME,
-} from "../../components/SortControl/SortControl";
+import SortControl from "../../components/SortControl/SortControl";
 
 import "./MovieListPage.css";
 import MovieTile from "../../components/MovieTile/MovieTile";
 import Search from "../../components/Search/Search";
-import MovieDetails from "../../components/MovieDetails/MovieDetails";
+
 import HeaderImg from "../../img/header.png";
 import { getMovie } from "../../api/api";
+import { useNavigate, useSearchParams } from "react-router-dom";
 
 const MovieListPage = () => {
-  const [searchQuery, setSearchQuery] = useState("");
-  const [sortCriterion, setSortCriterion] = useState(TYPE_SORT_NAME.title);
-
-  const [activeGenre, setActiveGenre] = useState("All");
+  let navigate = useNavigate();
 
   const [movieList, setMovieList] = useState(FILMS_DATA);
-  const [selectedMovie, setSelectedMovie] = useState(undefined);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const searchQuery = searchParams.get("query");
+  const sortCriterion = searchParams.get("sortBy");
+  const activeGenre = searchParams.get("genre");
 
   useEffect(() => {
     getMovie(sortCriterion, activeGenre, searchQuery)
       .then(function (response) {
-        setMovieList(response.data.data);
+        setMovieList(
+          response.data.data.length > 0 ? response.data.data : FILMS_DATA
+        );
+
+        FILMS_DATA.push(...response.data.data);
       })
       .catch(() => {
         setMovieList(FILMS_DATA);
       });
   }, [sortCriterion, activeGenre, searchQuery]);
+
+  const handleSearch = (value) => {
+    setSearchParams({ query: value });
+  };
+
+  const handleChangeSortControl = (event) => {
+    setSearchParams({ sortBy: event.target.value });
+  };
+  const handleChangeGenreSelect = (par) => {
+    setSearchParams({ genre: par });
+  };
```

Comment:
Revise these handlers. They should not conflict with each other. Each handler should be responsible for its own part (add/update).